intelligent_placer_lib/preprocessing.py

```diff
@@ -2,23 +2,42 @@
 import numpy as np
 from skimage.feature import canny
 
+def get_edges(image : np.ndarray) -> np.ndarray:
+    img = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    img = cv2.GaussianBlur(img, (3, 3), cv2.BORDER_DEFAULT)
+    img = canny(img).astype(np.uint8)
+    return img
 
-def preprocess_polygon(image):
+
+def preprocess_polygon(image : np.ndarray) -> np.ndarray:
     img = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
     img = cv2.GaussianBlur(img, (3, 3), cv2.BORDER_DEFAULT)
-    img = canny(img, sigma=1, low_threshold=0.1 * 255, high_threshold=0.2 * 255).astype(np.uint8)
+    img = canny(img).astype(np.uint8)
+
     img = cv2.morphologyEx(img, cv2.MORPH_CLOSE, np.ones((7, 7), dtype=np.uint8))
     img = cv2.morphologyEx(img, cv2.MORPH_OPEN, np.ones((3, 3), dtype=np.uint8))
     return img
 
 
-def preprocess_things(image):
+def preprocess_things(image : np.ndarray) -> np.ndarray:
     img = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
     img = cv2.GaussianBlur(img, (3, 3), cv2.BORDER_DEFAULT)
-    img = canny(img, sigma=3, low_threshold=10, high_threshold=30).astype(np.uint8)
+    img = canny(img).astype(np.uint8)
 
     kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (4, 4))
     img = cv2.dilate(img, kernel, iterations=3)
     img = cv2.erode(img, kernel, iterations=3)
 
     return img
+
+
+ # находим нижнюю границу листа, чтобы в дальнейшем разбить изображение на две половины
+def get_paper_line(image : np.ndarray) -> int:
+    img = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    img = cv2.GaussianBlur(img, (3, 3), cv2.BORDER_DEFAULT)
+    img = canny(img).astype(np.uint8)
```

Comment:
```suggestion
    img = get_edges(image)
```
То же в функциях `preprocess_things` и  `preprocess_polygon`

Reply:
Если я правильно понял, что вы имели в виду, то я осознанно добавил get_edges, но не стал её использовать)
Ибо всё-таки, вдруг может произойти такая ситуация, что параметры или вообще сам алгоритм выделения контуров  в преобработке для предметов и многоугольника могут быть различными, тогда одной функией неудобно будет описывать